test/QueryPlannerTestHelpers.h

```diff
@@ -420,6 +421,9 @@ constexpr auto OrderBy = [](const ::OrderBy::SortedVariables& sortedVariables,
 // Match a `UNION` operation.
 constexpr auto Union = MatchTypeAndOrderedChildren<::Union>;
 
+// Match a `DISTINCT` operation.
+constexpr auto Distinct = MatchTypeAndOrderedChildren<::Distinct>;
```

Comment:
The DISTINCT class also has the state of the selected/relevant variables.
Those should also be reflected in the matcher appropriately.